### PR TITLE
ulogd: init at 2.0.8

### PR DESCRIPTION
**Pull request by p-h**
###### Description of changes
- ulogd: init at 2.0.8
- linuxdoc-tools: init at 0.9.82
- ulogd service
- ulogd tests

###### Things done

This is almost entirely based on @xvuko's [PR](https://github.com/NixOS/nixpkgs/pull/103804), which sadly never got merged.
I rebased his branch on master and fixed the tests and updated and fixed the packages.

I'm unsure if I should squash anything or include him as a maintainer as he doesn't seem to have contributed anything since.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by RaitoBezarius** [comment]
Thank you for your contribution, you should remove the commits from @xvuko as the person may not want to be a maintainer or have anything to do with NixOS anymore.

File: nixos/modules/services/logging/ulogd.nix
Comment (RaitoBezarius) [inline nixos/modules/services/logging/ulogd.nix:43]: is `--verbose` something that a user would like to override?

Comment (p-h) [reply nixos/modules/services/logging/ulogd.nix:43]: There won't be any output in the journald log of the unit. I don't know if that is something anyone would want. Verbose isn't particularly verbose.

Comment (RaitoBezarius) [reply nixos/modules/services/logging/ulogd.nix:43]: Got it

File: nixos/modules/services/logging/ulogd.nix
Comment (RaitoBezarius) [inline nixos/modules/services/logging/ulogd.nix:36]: `with pkgs;` seems useless here

File: pkgs/tools/text/sgml/linuxdoc-tools/default.nix
Comment (RaitoBezarius) [inline pkgs/tools/text/sgml/linuxdoc-tools/default.nix:21]:
```suggestion
  LEX = "flex";
```

File: pkgs/tools/text/sgml/linuxdoc-tools/default.nix
Comment (RaitoBezarius) [inline pkgs/tools/text/sgml/linuxdoc-tools/default.nix:25]: I'm not sure to see why this is necessary, is it due to the PERL5LIB thing?

File: pkgs/tools/text/sgml/linuxdoc-tools/default.nix
Comment (RaitoBezarius) [inline pkgs/tools/text/sgml/linuxdoc-tools/default.nix:35]:
```suggestion
      ${lib.optionalString (!withLatex) "--replace '.tex .dvi .ps .pdf' ''"}
```

File: pkgs/tools/text/sgml/linuxdoc-tools/default.nix
Comment (RaitoBezarius) [inline pkgs/tools/text/sgml/linuxdoc-tools/default.nix:58]: Can you put a list of the licenses?

File: pkgs/os-specific/linux/ulogd/default.nix
Comment (RaitoBezarius) [inline pkgs/os-specific/linux/ulogd/default.nix:64]:
```suggestion
            The received messages can be logged into files or into a MySQL, SQLite 3
```

File: nixos/tests/ulogd.nix
Comment (RaitoBezarius) [inline nixos/tests/ulogd.nix:73]:
```suggestion
```

File: nixos/tests/ulogd.nix
Comment (RaitoBezarius) [inline nixos/tests/ulogd.nix:65]:
```suggestion
```

File: nixos/tests/ulogd.nix
Comment (RaitoBezarius) [inline nixos/tests/ulogd.nix:64]: Can we assert those are indeed ICMP?

File: nixos/tests/ulogd.nix
Comment (RaitoBezarius) [reply nixos/tests/ulogd.nix:65]: Useful for debugging, but I don't think we need it in the final test

**Review by RaitoBezarius — CHANGES_REQUESTED** [review]
Some minor stuff, but amazing work! Thank you a lot, I needed something like this!

**Comment by RaitoBezarius** [comment]
Also, can you separate the commit when you upgrade the package following CONTRIBUTING guidelines, i.e. `ulogd: 2.0.7 -> 2.0.8` from the commit where you become a maintainer (even better if you squash them IMHO so it becomes just an init at 2.0.8 and you were maintainer from the start.)

File: pkgs/tools/text/sgml/linuxdoc-tools/default.nix
Comment (p-h) [reply pkgs/tools/text/sgml/linuxdoc-tools/default.nix:27]: I don't like the look of this either. It seems to both require certain binaries in it's path as well as some of the perl libraries which it ships. The perl libraries seem to be installed in /share/linuxdoc-tools for some reason, which doesn't seem right to me.

Comment (p-h) [reply pkgs/tools/text/sgml/linuxdoc-tools/default.nix]: I reduced the amount of things added to the path, but I couldn't fix the rest.

File: pkgs/tools/text/sgml/linuxdoc-tools/default.nix
Comment (RaitoBezarius) [inline pkgs/tools/text/sgml/linuxdoc-tools/default.nix:60]: How did you derive these licenses?
Based on https://gitlab.com/agmartin/linuxdoc-tools/-/blob/master/COPYING ; I see MIT and various stuff that I am not trivially able to map on free.

File: nixos/tests/ulogd.nix
Comment (RaitoBezarius) [inline nixos/tests/ulogd.nix:63]: How is this sleep going to fare under a heavy load machine such as the Hydra builders?
Do we have any more reliable to wait for the ulogd to be filled with some timeout?

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 207453` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>linuxdoc-tools</li>
    <li>ulogd</li>
  </ul>
</details>

**Review by RaitoBezarius — APPROVED** [review]
LGTM, I am a bit afraid on the potential flakiness of the test due to sleep and licenses stuff.

File: nixos/tests/ulogd.nix
Comment (p-h) [reply nixos/tests/ulogd.nix:63]: > potential flakiness of the test due to sleep
Maybe stopping the daemon after the test before reading might work

> licenses stuff

linuxdoc-tools is also licensed partially under the sgmls licenseand the iso-entities license. I don't know how else to model this.

Comment (p-h) [reply nixos/tests/ulogd.nix:63]: Stopping doesn't work either. Some packages seem to be missed then as well.

Comment (RaitoBezarius) [reply nixos/tests/ulogd.nix:63]: `[ sgmls-license iso-entities-license ]` (replacing with their proper standard identifiers) is fine IMHO

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 207453` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>linuxdoc-tools</li>
    <li>ulogd</li>
  </ul>
</details>

File: nixos/tests/ulogd.nix
Comment (p-h) [reply nixos/tests/ulogd.nix:63]: Ah, I now see that iso-entities and entity-map are just licensed under MIT. I cannot find out what the SGML parser is licensed under. It seems to be its own license and [gentoo seems to treat it as its own license too](https://gitweb.gentoo.org/repo/gentoo.git/tree/licenses/SGMLUG?id=7d999af4a47bf55e53e54713d98d145f935935c1)

Comment (RaitoBezarius) [reply nixos/tests/ulogd.nix:63]: Awesome!

File: lib/licenses.nix
Comment (RaitoBezarius) [inline lib/licenses.nix:844]:
```suggestion
# Gentoo seems to treat it as a license: https://gitweb.gentoo.org/repo/gentoo.git/tree/licenses/SGMLUG?id=7d999af4a47bf55e53e54713d98d145f935935c1
```

File: pkgs/os-specific/linux/ulogd/default.nix
Comment (RaitoBezarius) [inline pkgs/os-specific/linux/ulogd/default.nix:52]:
```suggestion
passthru.tests = { inherit (nixosTest) ulogd; }
```

so we can detect breakages when bumping this package

File: pkgs/os-specific/linux/ulogd/default.nix
Comment (RaitoBezarius) [inline pkgs/os-specific/linux/ulogd/default.nix:3]:
```suggestion
, pkg-config, libpcap, linuxdoc-tools, autoreconfHook, nixosTests }:
```

File: lib/licenses.nix
Comment (RaitoBezarius) [inline lib/licenses.nix:844]: if you can insert a newline and insert a space, it would be awesome

**Review by RaitoBezarius — APPROVED** [review]
LGTM once minor nitpicks are addressed.

**Comment by RaitoBezarius** [comment]
I would wait for @alyssais to give a sanity check for the licenses' thing as they are code owner.

File: pkgs/tools/text/sgml/linuxdoc-tools/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/text/sgml/linuxdoc-tools/default.nix:12]:
```suggestion
  pname = "linuxdoc-tools";
  version = "0.9.82";

  src = fetchFromGitLab {
    owner = "agmartin";
    repo = "linuxdoc-tools";
    rev = version;
```

File: pkgs/tools/text/sgml/linuxdoc-tools/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/text/sgml/linuxdoc-tools/default.nix:21]:
```suggestion
  configureFlags = [
    ("--enable-docs=txt info lyx html rtf"
      + lib.optionalString withLatex " pdf")
  ];
```

File: pkgs/tools/text/sgml/linuxdoc-tools/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/text/sgml/linuxdoc-tools/default.nix:42]:
```suggestion
  nativeBuildInputs = [ flex which makeWrapper ];
```
part of stdenv

File: pkgs/tools/text/sgml/linuxdoc-tools/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/text/sgml/linuxdoc-tools/default.nix:58]:
```suggestion
    longDescription = ''
      A collection of text formatters which understands a LinuxDoc DTD SGML
      source file. Each formatter (or "back-end") renders the source file into
      a variety of output formats, including HTML, TeX, DVI, PostScript, plain
      text, and groff source in manual-page format. The linuxdoc suite is
      provided for backward compatibility, because there are still many useful
      documents written in LinuxDoc DTD sgml source.
    '';
```

File: pkgs/os-specific/linux/ulogd/default.nix
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/ulogd/default.nix:44]:
```suggestion
```
already in stdenv

File: pkgs/os-specific/linux/ulogd/default.nix
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/ulogd/default.nix:16]:
```suggestion
  postPatch = ''
```

File: nixos/modules/services/logging/ulogd.nix
Comment (SuperSandro2000) [inline nixos/modules/services/logging/ulogd.nix:46]: systemd should natively support this

File: nixos/modules/services/logging/ulogd.nix
Comment (SuperSandro2000) [inline nixos/modules/services/logging/ulogd.nix:32]:
```suggestion
        description = lib.mdDoc "Configuration for ulogd. See {file}`/share/doc/ulogd/` in `pkgs.ulogd.doc`.";
      };

      logLevel = mkOption {
        type = types.enum [ 1 3 5 7 8 ];
        default = 5;
        description = lib.mdDoc "Log level (1 = debug, 3 = info, 5 = notice, 7 = error, 8 = fatal)";
```

File: nixos/modules/services/logging/ulogd.nix
Comment (SuperSandro2000) [inline nixos/modules/services/logging/ulogd.nix:25]: Is there an online doc we could link?

File: nixos/modules/services/logging/ulogd.nix
Comment (p-h) [reply nixos/modules/services/logging/ulogd.nix:44]: What do you mean by this? I read systemd.service(5) and it only seems to suggest that using SIGHUP to trigger a reload might not be the best idea, but nothing else. Removing the line also makes the reload test fail.

File: nixos/modules/services/logging/ulogd.nix
Comment (p-h) [reply nixos/modules/services/logging/ulogd.nix]: Nothing official really, except [this](https://www.netfilter.org/projects/ulogd/index.html). The documentation shipped with the package is hosted on third-party sites, if that is what you want.

**Comment by p-h** [comment]
#103804 can probably be closed now

**Pull request merged**
